### ISSUE TITLE
Redundant Code in device_tracer_cuda.cc

### ISSUE THREAD

**Issue by yezhengmao1** [openxla/xla#20619]
https://github.com/openxla/xla/blob/26ba40164b5608c18c0747cffb73563caf53dfc4/xla/backends/profiler/gpu/device_tracer_cuda.cc#L230

It seems that when constructing the GpuTrace, the pointer of cupti_interface does not need to be used.

in line 225, will call GetCuptiTracerSignleton to create it.

**Comment by akuegel** [comment]
Yes, you are right, there is a possibility for code cleanup here. Thanks for pointing it out.

**Comment by akuegel** [comment]
Fixed with https://github.com/openxla/xla/commit/7a801635665be5c7208b7271a001d4d6596e56ff
